src/create-theme-provider.js

```diff
@@ -8,7 +8,7 @@ import isObject from './is-object';
 
 export type ThemeProviderProps<Theme> = {
   children: Node,
-  theme: Theme | (outerTheme: Theme) => Theme,
+  theme: $NonMaybeType<Theme> | (outerTheme: Theme) => $NonMaybeType<Theme>,
```

Comment:
`theme: Theme | (outerTheme: Theme) => Theme` would allow for `null` to be passed to the `ThemeProvider` as a prop or return it from the `function`.

Reply:
but you haven't defined Theme as Maybe type anywere, or do I miss it?

Reply:
Well, our default theme is defined as `{} | null`. But in the case when some create their ThemeProvider, this will also not allow passing in a `null` or `undefined` value.

Reply:
mb we should ask for a single consistent value and thats it, idk, using in one case a maybe type and then fixing it with $NonMaybeType sounds like a workaround

Reply:
Well, we need to use `{} | null` to not merge the theme when there is no outer theme.

And this will give users a warning when they try to pass a maybe type.

Reply:
why not `{} | undefined` ?

Reply:
Why would the theme prop be optional on `ThemeProvider`? This would also allow to return `undefined` from the function.

And this wouldn't do that check in custom created ThemeProviders.

Reply:
I was talking about the Context, in Context theme is optional, can be object or missing, right?

Reply:
yeah, in the context type it's either an object or null